File: front/src/types/order.ts
export type OrderStatus = "Created" | "Completed";

export interface IOrderItemDTO {
  id: string;
  mealId: string;
  mealName: string;
  quantity: number;
  unitPrice: number;
  note: string | null;
}

export interface IOrderDTO {
  id: string;
  orderNumber: number;
  status: OrderStatus;
  items: Array<IOrderItemDTO>;
  total: number;
  discount: number;
  isPaid: boolean;
  isDineIn: boolean;
  deliveryAddress: string | null;
  createdAt: string;
  updatedAt: string;
}

File: back/FoodEpos/Model/Order.cs
namespace FoodEpos.API.Model
{
    public class Order
    {
        public Guid OrderId { get; set; }
        public int OrderNum { get; set; }
        public List<OrderItems> OrderItems { get; set; } = [];
        public float Total { get; set; }
        public float Discount { get; set; }
        public bool IsPaid { get; set; }
        public bool IsDineIn { get; set; }
        public string? DeliveryAddress { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
